src/main/java/com/google/cloud/tools/managedcloudsdk/install/Extractor.java

```diff
@@ -42,24 +44,61 @@
     this.messageListener = messageListener;
   }
 
-  /** Extract and return a {@link Path} to the Cloud SDK home directory. */
-  @Override
-  public Path call() throws IOException {
+  /** Extract and return a {@link Path} to the extraction root. */
+  public Path extract() throws IOException, InterruptedException {
     messageListener.message("Extracting archive: " + archive);
 
-    extractorProvider.extract(archive, destination, messageListener);
+    try {
+      extractorProvider.extract(archive, destination, messageListener);
+    } catch (IOException ex) {
+      try {
+        messageListener.message("Extraction failed, cleaning up " + destination);
+        cleanUp(destination);
+      } catch (Exception exx) {
```

Comment:
Should this be just catching an IOException?

Reply:
Yeah, probably.